client/src/utils/signUpValidation.js

```diff
@@ -0,0 +1,9 @@
+import * as Yup from 'yup';
+const validationSignUpSchema = Yup.object({
+  username:Yup.string().min(3,'Username should be at least 3 char ').max(20,'Username should be at most 20 char ').required('username'),
```

Comment:
you should be consistent, so please instead of .required(username)
.required('username is required')